src/Sarif.Multitool/FileWorkItemsCommand.cs

```diff
@@ -28,89 +35,56 @@ public int Run(FileWorkItemsOptions options)
-                        logsToProcess = visitor.SplitSarifLogs;
-                    }
-
-                    IList<WorkItemModel> workItemModels = new List<WorkItemModel>(logsToProcess.Count);
-
-                    for (int splitFileIndex = 0; splitFileIndex < logsToProcess.Count; splitFileIndex++)
-                    {
-                        SarifLog splitLog = logsToProcess[splitFileIndex];
-                        WorkItemModel workItemModel = splitLog.CreateWorkItemModel(filingClient.ProjectOrRepository);
-                        workItemModels.Add(workItemModel);
-                    }
-
-                    try
-                    {
-                        string securityToken = Environment.GetEnvironmentVariable("SarifWorkItemFilingSecurityToken");
-
-                        if (string.IsNullOrEmpty(securityToken))
-                        {
-                            throw new InvalidOperationException("'SarifWorkItemFilingSecurityToken' environment variable is not initialized with a personal access token.");
-                        }
-
-                        IEnumerable<WorkItemModel> filedWorkItems = filer.FileWorkItems(options.ProjectUri, workItemModels, securityToken).Result;
-                        Console.WriteLine($"Created {filedWorkItems.Count()} work items for run {runIndex}.");
-                    }
-                    catch (Exception ex)
-                    {
-                        Console.Error.WriteLine(ex);
-                    }
-                }
+                filingContext.DataToInsert = options.DataToInsert.ToFlags();
             }
 
-            Console.WriteLine($"Writing log with work item Ids to {options.OutputFilePath}.");
-            WriteSarifFile<SarifLog>(fileSystem, sarifLog, options.OutputFilePath, (options.PrettyPrint ? Formatting.Indented : Formatting.None));
+            var filer = new SarifWorkItemFiler(filingContext);
+            filer.FileWorkItems(logFileContents);
+
+            // TODO: We need to process updated work item models to persist filing
+            //       details back to the input SARIF file, if that was specified.
+            //       The SarifWorkItemFi;er should either return or persist the updated
+            //       models via a property, so that we can do this work.
+            //
+            //       This information should be inlined to the input file, if configured,
+            //       or persisted to a new SARIF file, if configured. If neither of 
+            //       those options is specified, there is no work to do. 
+            //
+            //       https://github.com/microsoft/sarif-sdk/issues/1774
 
             return SUCCESS;
         }
 
-        private bool ValidateOptions(FileWorkItemsOptions options, IFileSystem fileSystem)
+        private bool ValidateOptions(FileWorkItemsOptions options, SarifWorkItemContext workItemFilingConfiguration, IFileSystem fileSystem)
```

Comment:
>workItemFilingConfiguration [](start = 88, length = 27)

Did you mean to name this `workItemFilingContext` (or just `filingContext`)? It's true that it starts out as the contents of the config file, but that's not how it ends up -- and the caller uses `filingContext`.

Reply:
Just a missed rename oppt'y. Thanks for pointing it out.

---
In reply to: [381018380](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381018380) [](ancestors = 381018380)